quantum/quantum.c

```diff
@@ -481,3 +481,39 @@ void api_send_unicode(uint32_t unicode) {
 __attribute__((weak)) void startup_user() {}
 
 __attribute__((weak)) void shutdown_user() {}
+
+// Useful string manipulation code
+const char *get_u8_str(uint8_t curr_num, char curr_pad) {
+    static char    buf[4]   = {0};
+    static uint8_t last_num = 0xFF;
+    static char    last_pad = '\0';
+    if (last_num == curr_num && last_pad == curr_pad) {
+        return buf;
+    }
+    last_num = curr_num;
+    last_pad = curr_pad;
+    buf[3]   = '\0';
+    buf[2]   = '0' + curr_num % 10;
+    buf[1]   = (curr_num /= 10) % 10 ? '0' + (curr_num) % 10 : (curr_num / 10) % 10 ? '0' : curr_pad;
+    buf[0]   = curr_num / 10 ? '0' + curr_num / 10 : curr_pad;
+    return buf;
+}
+
+const char *get_u16_str(uint16_t curr_num, char curr_pad) {
+    static char     buf[6]   = {0};
+    static uint16_t last_num = 0xFF;
+    static char     last_pad = '\0';
```

Comment:
```suggestion
    static char     buf[6]   = {0};
    static uint16_t last_num = 0;
    static char     last_pad = '\0';
```
The same change as above in `get_u8_str()` — `last_pad = '\0'` is invalid, therefore the initial value of `last_num` does not matter, and a zero-initialized static variable can go into BSS, saving 2 bytes in this case.